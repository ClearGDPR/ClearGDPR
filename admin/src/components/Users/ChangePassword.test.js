import React from 'react';
import { shallow, mount } from 'enzyme';

import ChangePasswordForm, { ChangePassword } from './ChangePassword';

const setupShallow = propOverrides => {
  const props = Object.assign({}, propOverrides);
  const component = shallow(<ChangePassword {...props} />);

  return { props, component };
};

const setupMount = propOverrides => {
  const props = Object.assign({}, propOverrides);
  const component = mount(<ChangePasswordForm {...props} />);

  return { props, component };
};

describe('(Component) Change Password', () => {
  it('should render correctly when no props provided', async () => {
    const { component } = setupShallow();
    expect(component).toMatchSnapshot();
  });

  it('should render correctly when the prop isLoading is set to false', async () => {
    const { component } = setupShallow({ isLoading: false });
    expect(component).toMatchSnapshot();
  });

  it('should render correctly when the prop isLoading is set to true', async () => {
    const { component } = setupShallow({ isLoading: true });
    expect(component).toMatchSnapshot();
  });

  it('should validate password when inputs values change', async () => {
    // TODO: refactor 
  });

  it('should submit when validation is ok', async () => {
    // TODO: refactor
  });

  it('should render correct props when touched', async () => {
    const { component } = setupShallow({
      isLoading: false,
      touched: { newPassword: true }
    });

    const newPasswordInput = component.find('TextInput[field="newPassword"]').at(0);
    expect(newPasswordInput.props().error).toBeFalsy();
  });

  it('should render correct props when touched and there are errors', async () => {
    const { component } = setupShallow({
      isLoading: false,
      touched: { newPassword: true },
      errors: { newPassword: 'Error message' }
    });

    const newPasswordInput = component.find('TextInput[field="newPassword"]').at(0);

    expect(newPasswordInput.props().error).toEqual('Error message');
  });
});
