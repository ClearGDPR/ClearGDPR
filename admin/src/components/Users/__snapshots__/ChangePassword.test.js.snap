// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Change Password should render correctly when no props provided 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChangePassword />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <React.Fragment>
        <TextInput
          field="newPassword"
          label="New password"
          placeholder="*********"
          type="password"
          validate={[Function]}
        />
        <TextInput
          field="newPasswordRepeat"
          label="Repeat new password"
          placeholder="*********"
          type="password"
          validate={[Function]}
        />
        <div>
          <input
            className="btn"
            type="submit"
            value="Save"
          />
        </div>
      </React.Fragment>,
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <TextInput
            field="newPassword"
            label="New password"
            placeholder="*********"
            type="password"
            validate={[Function]}
          />,
          <TextInput
            field="newPasswordRepeat"
            label="Repeat new password"
            placeholder="*********"
            type="password"
            validate={[Function]}
          />,
          <div>
            <input
              className="btn"
              type="submit"
              value="Save"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "error": undefined,
            "field": "newPassword",
            "label": "New password",
            "placeholder": "*********",
            "type": "password",
            "validate": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "error": undefined,
            "field": "newPasswordRepeat",
            "label": "Repeat new password",
            "placeholder": "*********",
            "type": "password",
            "validate": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="btn"
              type="submit"
              value="Save"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "btn",
              "type": "submit",
              "value": "Save",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <React.Fragment>
          <TextInput
            field="newPassword"
            label="New password"
            placeholder="*********"
            type="password"
            validate={[Function]}
          />
          <TextInput
            field="newPasswordRepeat"
            label="Repeat new password"
            placeholder="*********"
            type="password"
            validate={[Function]}
          />
          <div>
            <input
              className="btn"
              type="submit"
              value="Save"
            />
          </div>
        </React.Fragment>,
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <TextInput
              field="newPassword"
              label="New password"
              placeholder="*********"
              type="password"
              validate={[Function]}
            />,
            <TextInput
              field="newPasswordRepeat"
              label="Repeat new password"
              placeholder="*********"
              type="password"
              validate={[Function]}
            />,
            <div>
              <input
                className="btn"
                type="submit"
                value="Save"
              />
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "error": undefined,
              "field": "newPassword",
              "label": "New password",
              "placeholder": "*********",
              "type": "password",
              "validate": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "error": undefined,
              "field": "newPasswordRepeat",
              "label": "Repeat new password",
              "placeholder": "*********",
              "type": "password",
              "validate": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                className="btn"
                type="submit"
                value="Save"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "btn",
                "type": "submit",
                "value": "Save",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`(Component) Change Password should render correctly when the prop isLoading is set to false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChangePassword
    isLoading={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <React.Fragment>
        <TextInput
          field="newPassword"
          label="New password"
          placeholder="*********"
          type="password"
          validate={[Function]}
        />
        <TextInput
          field="newPasswordRepeat"
          label="Repeat new password"
          placeholder="*********"
          type="password"
          validate={[Function]}
        />
        <div>
          <input
            className="btn"
            type="submit"
            value="Save"
          />
        </div>
      </React.Fragment>,
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <TextInput
            field="newPassword"
            label="New password"
            placeholder="*********"
            type="password"
            validate={[Function]}
          />,
          <TextInput
            field="newPasswordRepeat"
            label="Repeat new password"
            placeholder="*********"
            type="password"
            validate={[Function]}
          />,
          <div>
            <input
              className="btn"
              type="submit"
              value="Save"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "error": undefined,
            "field": "newPassword",
            "label": "New password",
            "placeholder": "*********",
            "type": "password",
            "validate": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "error": undefined,
            "field": "newPasswordRepeat",
            "label": "Repeat new password",
            "placeholder": "*********",
            "type": "password",
            "validate": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="btn"
              type="submit"
              value="Save"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "btn",
              "type": "submit",
              "value": "Save",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <React.Fragment>
          <TextInput
            field="newPassword"
            label="New password"
            placeholder="*********"
            type="password"
            validate={[Function]}
          />
          <TextInput
            field="newPasswordRepeat"
            label="Repeat new password"
            placeholder="*********"
            type="password"
            validate={[Function]}
          />
          <div>
            <input
              className="btn"
              type="submit"
              value="Save"
            />
          </div>
        </React.Fragment>,
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <TextInput
              field="newPassword"
              label="New password"
              placeholder="*********"
              type="password"
              validate={[Function]}
            />,
            <TextInput
              field="newPasswordRepeat"
              label="Repeat new password"
              placeholder="*********"
              type="password"
              validate={[Function]}
            />,
            <div>
              <input
                className="btn"
                type="submit"
                value="Save"
              />
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "error": undefined,
              "field": "newPassword",
              "label": "New password",
              "placeholder": "*********",
              "type": "password",
              "validate": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "error": undefined,
              "field": "newPasswordRepeat",
              "label": "Repeat new password",
              "placeholder": "*********",
              "type": "password",
              "validate": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                className="btn"
                type="submit"
                value="Save"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "btn",
                "type": "submit",
                "value": "Save",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`(Component) Change Password should render correctly when the prop isLoading is set to true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChangePassword
    isLoading={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Loader />,
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Loader />,
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
